### PR TITLE
Improve `code-style` workflow by tweaking path filters

### PR DESCRIPTION
**Pull request by trsvchn**
Addresses #1836 

Description:

Replace `path-ignore` with `path` filters:

> A matching negative pattern (prefixed with !) after a positive match will exclude the path.

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by vfdev-5** [comment]
@trsvchn thanks for the PR, but why it is an improvement ?

**Comment by trsvchn** [comment]
@vfdev-5 Good question! I think in updated configuration this workflow starts only when py file was modifed, except specified dirs. And no startups on bash, yml, toml, or cfg updates

With old filters:

```yml
    paths-ignore:
      - "assets/**"
      - ".circleci/**"
      - "docker/**"
      - "docs/**"
      - "conda.recipe"
      - "**.md"
```

we trigger `code-style` checks on any kind of files, except `md` files + ignore some dirs. So, it is not python oriented, for example I can change some bash files in `tests` or `.github` dirs, or yaml files in `.github` etc. This PR is also an example.

For me, the main aim of this workflow is fixing only python files and ignore anything else.

If you prefer to keep `paths-ignore` style, we can try smth like:

```yml
    paths-ignore:
      - "assets/**"
      - ".circleci/**"
      - "docker/**"
      - "docs/**"
      - "conda.recipe"
      - "**.md"

      - ".github/**"
      - "**.sh"
```

What do you think?

File: .github/workflows/code-style.yml [hunk 1]
```diff
@@ -2,29 +2,32 @@ name: Format python code
...
-      - "assets/**"
-      - ".circleci/**"
-      - "docker/**"
-      - "docs/**"
-      - "conda.recipe"
-      - "**.md"
-  #   pull_request:
-  #     paths-ignore:
+    paths:
+      - "**.py"
```
Comment (trsvchn) [inline .github/workflows/code-style.yml:6]:
```suggestion
      - "**.py"
      - "setup.cfg"
      - ".github/workflows/code-style.yml"
```

Comment (trsvchn) [reply .github/workflows/code-style.yml:6]: Add some non-py files related to code-style checks

**Comment by vfdev-5** [comment]
Here is an example when we would like to trigger unit tests but modified only `requirements-dev.txt` and currently it does not run any tests: https://github.com/pytorch/ignite/pull/1863

**Comment by vfdev-5** [comment]
Finally I have read and understood your comment and your point. I agree with this approach. Let's finalize the PR and merge it

**Review by vfdev-5 — APPROVED** [review]
Thanks !

**Pull request merged**